### PR TITLE
Fix php 8.1 errors

### PR DESCRIPTION
**Pull request by andrew-cormick-dockery**
Tested with the User Story test suite.

### CODE REVIEW

File: foreigndonors.php [hunk 1]
```diff
@@ -125,23 +127,20 @@ function foreigndonors_civicrm_buildForm($formName, &$form) {
...
     elseif ($domainId == 9 || $isACT ) {
-      $form->assign('addACTProhibitedDonor', TRUE);
       $label = "I am an Australian Citizen or Permanent Resident and am not a foreign donor";
       $form->add('Checkbox', 'foreigndonor', ts('I am an Australian Citizen or Permanent Resident and am not a foreign donor'));
       $form->addRule('foreigndonor', ts('You must affirm you are an Australian Citizen or Permanent Resident'), 'required', NULL, 'client');
       $form->add('Checkbox', 'prohibiteddonor', ts('I am not a prohibited donor as per ACT electoral legislation'));
       $form->addRule('prohibiteddonor', ts('You must affirm you are not a prohibited donor as per ACT electoral legislation'), 'required', NULL, 'client');
     }
     else {
-      $form->assign('addProhibitedDonor', FALSE);
```
Comment (seamuslee001) [inline foreigndonors.php:144]: @andrew-cormick-dockery is this variable not used in the template?

File: foreigndonors.php [hunk 2]
```diff
@@ -115,6 +115,8 @@ function foreigndonors_civicrm_buildForm($formName, &$form) {
 
     $isNSW = substr($finType[0]['financial_type_id:label'], 0, 3) === 'NSW';
     $isACT = substr($finType[0]['financial_type_id:label'], 0, 3) === 'ACT';
+    $form->assign('addNSWProhibitedDonor', $isNSW);
+    $form->assign('addACTProhibitedDonor', $isACT);
```
Comment (seamuslee001) [inline foreigndonors.php]: @andrew-cormick-dockery don't you need to include the $domainId part of the if statement as well here?

File: foreigndonors.php (hunk 1 above)
Comment (andrew-cormick-dockery) [reply foreigndonors.php:144]: I can't see a reference anywhere. Maybe it used to be used but now is not?

Comment (seamuslee001) [reply foreigndonors.php:144]: yeh ok makes sense then

**Review by seamuslee001 — APPROVED** [review]

**Review by johntwyman — APPROVED** [review]

**Pull request merged**